sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriterV2.scala

```diff
@@ -167,6 +173,241 @@ final class DataFrameWriterV2[T] private[sql](table: String, ds: Dataset[T])
     runCommand(overwrite)
   }
 
+  /**
+   * Specifies the merge condition.
+   *
+   * Sets the condition, provided as a `String`, to be used for merging data. This condition
+   * is converted internally to a `Column` and used to determine how rows from the source
+   * DataFrame are matched with rows in the target table.
+   *
+   * @param condition a `String` representing the merge condition.
+   * @return the current `DataFrameWriterV2` instance with the specified merge condition set.
+   */
+  def on(condition: String): DataFrameWriterV2[T] = {
```

Comment:
Shall we add a new layer for merge APIs? e.g. `partitionedBy` returns `CreateTableWriter`

Reply:
Added `MergeIntoWriter[T]`